core/src/main/java/com/amplifyframework/core/model/ModelSchemaRegistry.java

```diff
@@ -57,8 +55,10 @@ public synchronized void load(@NonNull Set<Class<? extends Model>> models) throw
      *                        {@link Class#getSimpleName()} method.
      * @return the ModelSchema object for the given Model class.
      */
+    @NonNull
     public synchronized ModelSchema getModelSchemaForModelClass(@NonNull String classSimpleName) {
-        return modelSchemaMap.get(classSimpleName);
+        final ModelSchema result = modelSchemaMap.get(classSimpleName);
+        return Objects.requireNonNull(result);
```

Comment:
Do you think in this case it'd make sense to throw a custom error with an explanation if it is null since this isn't checking input provided by the customer but rather the result we're returning? Same for below on line 73?

Reply:
Hm, yes, I agree. I'll update this.